packages/envconfig/src/types.ts

```diff
@@ -0,0 +1,116 @@
+import type { NativeConnectionOptions } from '@temporalio/worker';
+
+/**
+ * A data source for configuration, which can be a path to a file,
+ * the string contents of a file, or raw bytes.
+ *
+ * @experimental Environment configuration is new feature and subject to change.
+ */
+export type ConfigDataSource = { path: string } | { data: string | Buffer };
```

Comment:
I'm hesitant here.

For one thing, `Buffer` is a Node-specific type, which will pose issues at some point in the future when we'll add official support for other JS runtimes. We can use that class as much as we want internally, because replacing it at a later time won't be a breaking change, but it was a mistake to ever use that type in our public APIs. That should be `Uint8Array` instead, which is part of the EcmaScript standard.

Second, I get the feeling that `data` being `string | Uint8Array` is adding complexity without any real benefit. Any reason not to make this `Uint8Array` only?

Reply:
I had switched it to use `Uint8Array` while making the change to make this standalone without core for this exact reason. But I noticed usage of `Buffer` in `tls-config.ts` and in the worker, so I figured that this was already a path walked and thought it'd be more consistent.

No issues changing to `Uint8Array`.